clang/include/clang/AST/Type.h

```diff
@@ -6155,6 +6155,53 @@ class BTFTagAttributedType : public Type, public llvm::FoldingSetNode {
   }
 };
 
+class HLSLAttributedResourceType : public Type, public llvm::FoldingSetNode {
+public:
+  struct Attributes {
+    // Data gathered from HLSL resource attributes
+    uint8_t ResourceClass; // maps to llvm::dxil::ResourceClass
+    uint8_t IsROV : 1;
+    Attributes(uint8_t RC, bool rov) : ResourceClass(RC), IsROV(rov) {}
+    Attributes() : Attributes(0, 0) {}
```

Comment:
Do we actually need the default constructor? I can't imagine we ever build an empty attributed resource type

Reply:
I want to make sure that the struct is always zero-initialized. Is there some attribute or trick I can add here to make sure of that, other than the default constructor? I know I can zero-initialize by assigning `= {}` when it is declared as variable and there is probably going to be just a single place where we need to do that so maybe this is not really necessary.

Reply:
If we don't ever default construct the object, this can simply be `Attributes() = delete;`, so trying to create this without initializing it is impossible. If we do need a default constructor then what you have is the best we can do - default member initialization for bitfields is in C++20 but we have to stick to 17.